src/Mod/Sketcher/App/SketchObject.cpp

```diff
@@ -4626,11 +4626,16 @@ bool SketchObject::isCarbonCopyAllowed(App::Document* pDoc, App::DocumentObject*
     if (rsn)
         *rsn = rlAllowed;
 
+    std::string sketchArchType ("Sketcher::SketchObjectPython");
+
     // Only applicable to sketches
-    if (pObj->getTypeId() != Sketcher::SketchObject::getClassTypeId()) {
-        if (rsn)
-            *rsn = rlNotASketch;
-        return false;
+    if (pObj->getTypeId() != Sketcher::SketchObject::getClassTypeId() ) {
+        if (sketchArchType.compare(pObj->getTypeId().getName()) != 0) {

```

Comment:
How about combining these into one if instead?
Also, you can use == to compare strings